interfaces: add udev netlink support to hardware-observe

mvo5 opened the pull request:
With the addition of seccomp argument filtering the socket syscall
will not allow NETLINK_KOBJECT_UEVENT anymore. But this is crucial
for the hardware-observe interface to monitor udev. This patch
adds it back.

jdstrand commented on interfaces/builtin/hardware_observe.go:
```diff
@@ -75,6 +75,9 @@ capability sys_admin,
 @{PROC}/sys/abi/{,*} r,
+
+# for udev
+network netlink raw,
```
I would move this up to under the udevadm rule.

zyga commented:
I would love to have not only a description of what the change is (especially for a trivial patch that can be read in a second) but more importantly *why* we are making that change.

jdstrand commented:
@zyga - in this case I think it is clear from the description. hardware-observe simply should've always had udev netlink support.

jdstrand commented on interfaces/builtin/hardware_observe.go:
```diff
@@ -87,6 +90,9 @@ iopl
 socket AF_NETLINK - NETLINK_GENERIC
+
+# kernel uevents
+socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
```
Please also add `bind` here.

mvo5 replied:
Added

mvo5 commented:
I addressed the feedback and also improved the commit message. I had to force push for this, sorry for that, however the upside of that is that I can easily create 2.27 and 2.28 branches from this commit.
